frontend/packages/console-shared/src/components/dynamic-form/fields.tsx

```diff
@@ -26,75 +30,144 @@ export const DescriptionField: React.FC<FieldProps> = ({ id, description }) =>
-  return (
-    <FieldSet {...props}>
-      <dl id={idSchema.$id} style={{ marginLeft: '15px' }}>
-        <dt>Limits</dt>
-        <dd>
-          <ResourceRequirements
-            cpu={formData?.limits?.cpu || ''}
-            memory={formData?.limits?.memory || ''}
-            storage={formData?.limits?.['ephemeral-storage'] || ''}
-            onChangeCPU={(cpu) => onChange(_.set(_.cloneDeep(formData), 'limits.cpu', cpu))}
-            onChangeMemory={(mem) => onChange(_.set(_.cloneDeep(formData), 'limits.memory', mem))}
-            onChangeStorage={(sto) =>
-              onChange(_.set(_.cloneDeep(formData), 'limits.ephemeral-storage', sto))
-            }
-            path={`${idSchema.$id}.limits`}
-          />
-        </dd>
-        <dt>Requests</dt>
-        <dd>
-          <ResourceRequirements
-            cpu={formData?.requests?.cpu || ''}
-            memory={formData?.requests?.memory || ''}
-            storage={formData?.requests?.['ephemeral-storage'] || ''}
-            onChangeCPU={(cpu) => onChange(_.set(_.cloneDeep(formData), 'requests.cpu', cpu))}
-            onChangeMemory={(mem) => onChange(_.set(_.cloneDeep(formData), 'requests.memory', mem))}
-            onChangeStorage={(sto) =>
-              onChange(_.set(_.cloneDeep(formData), 'requests.ephemeral-storage', sto))
-            }
-            path={`${idSchema.$id}.requests`}
-          />
-        </dd>
-      </dl>
-    </FieldSet>
+export const FieldSet: React.FC<FieldSetProps> = ({
+  children,
+  defaultLabel,
+  idSchema,
+  required = false,
+  schema,
+  uiSchema,
+}) => {
+  const [expanded, setExpanded] = React.useState(false);
+  const [showLabel, label] = useSchemaLabel(schema, uiSchema, defaultLabel);
+  const description = useSchemaDescription(schema, uiSchema);
+  const onToggle = (e) => {
+    e.preventDefault();
+    setExpanded((current) => !current);
+  };
+  return showLabel && label ? (
+    <div id={`${idSchema.$id}_field-group`} className="form-group co-dynamic-form__field-group">
+      <AccordionItem>
+        <AccordionToggle
+          id={`${idSchema.$id}_accordion-toggle`}
+          onClick={onToggle}
+          isExpanded={expanded}
+        >
+          <label
+            className={classnames({ 'co-required': required })}
+            htmlFor={`${idSchema.$id}_accordion-content`}
+          >
+            {_.startCase(label)}
+          </label>
+        </AccordionToggle>
+        {description && (
+          <Description id={`${idSchema.$id}_description`} description={description} />
+        )}
+        <AccordionContent id={`${idSchema.$id}_accordion-content`} isHidden={!expanded}>
+          {children}
+        </AccordionContent>
+      </AccordionItem>
+    </div>
+  ) : (
+    <>{children}</>
   );
 };
 
-export const UpdateStrategyField: React.FC<FieldProps> = (props) => {
-  const { formData = {}, idSchema, onChange } = props;
+export const ResourceRequirementsField: React.FC<FieldProps> = ({
```

Comment:
This just got pushed down because of the other components that were added at the top of the file. The rest of the changes in this file were mostly just getting rid of the spread operator where `FieldSet` is used and making better use of the null coalescence operator.